### PR TITLE
add xds proxy metrics

### PR DESCRIPTION
**Pull request by ramaraochavali**
Adds metrics for some xds proxy operations that help in debugging.
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/istio-agent/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	istiodDisconnections = monitoring.NewSum(
+		"istiod_connection_terminations",
+		"The total number of connection errors to Istiod",
+		monitoring.WithLabels(disconnectionTypeTag),
+	)
+
+	// envoyDisconnections records total number of unexpected disconnections by Envoy.
+	envoyDisconnections = monitoring.NewSum(
+		"istiod_connection_terminations",
+		"The total number of connection errors to Istiod",
```
Comment (howardjohn) [inline pkg/istio-agent/metrics/metrics.go:40]: description is wrong

**Review by howardjohn — COMMENTED** [review]
LGTM other than minor comment

File: pkg/istio-agent/metrics/metrics.go (hunk 1 above)
Comment (ramaraochavali) [reply pkg/istio-agent/metrics/metrics.go]: Ah. thanks. Fixed now

**Review by howardjohn — APPROVED** [review]

**Pull request merged**